src/modal/alert-modal.component.ts

```diff
@@ -0,0 +1,138 @@
+import {
+	Component,
+	Injector,
+	OnInit,
+	ElementRef,
+	AfterViewInit
+} from "@angular/core";
+import {
+	trigger,
+	state,
+	style,
+	transition,
+	animate
+} from "@angular/animations";
+import Modal from "./modal.decorator";
+import { ModalService } from "./modal.service";
+
+/**
+ * Component to create standard modals for presenting content or asking for user's input.
+ * It can show as a passive modal showing only text or show as a transactional modal with
+ * multiple buttons for different actions for the user to choose from.
+ *
+ * Using a modal in your application requires `ibm-modal-placeholder` which would generally be
+ * placed near the end of your app component template (app.component.ts or app.component.html) as:
+ *
+ * ```html
+ * <ibm-modal-placeholder></ibm-modal-placeholder>
+ * ```
+ *
+ * Example of opening the modal:
+ *
+ * ```typescript
+ * \@Component({
+ *  selector: "app-modal-demo",
+ *  template: `
+ *   <button class="btn--primary" (click)="openModal()">Open modal</button>
+ *   <ibm-modal-placeholder></ibm-modal-placeholder>`
+ * })
+ * export class ModalDemo {
+ *
+ * 	openModal() {
+ * 		this.modalService.show({
+ *			modalType: "default" | "danger",
+ *			headerText: "optional header text",
```

Comment:
I'm thinking `headerText` should be `headerLabel` since that's actually where that text outputs to

Reply:
sure, I see that they named "Optional label" on Carbon's website. I wish I could explain where did I see "header text" :D